qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+T1 Experiment class.
+"""
+
+from typing import List, Optional, Union, Tuple
+import numpy as np
+from scipy.optimize import curve_fit
+
+from qiskit.circuit import QuantumCircuit
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments import AnalysisResult
+
+
+class T1Analysis(BaseAnalysis):
+    """T1 Experiment result analysis class."""
+
+    def _run_analysis(self, experiment_data, **params) -> Tuple[AnalysisResult, None]:
+        """
+        Calculate T1
+
+        Args:
+            experiment_data: the experiment data to analyze
+            params: expected parameters therein are:
+                   `fit_p0` and `fit_bounds` - to be passed to scipy.optimize.curve_fit
+                                               as the `p0` and `bounds` parameters
+
+        Returns:
+            The analysis result with the estimated T1
+        """
+
```

Comment:
Chris: I think having a generic curve fit, exp fit, etc functions / analysis that can be called / subclassed makes sense for these things.

Reply:
I externalized `exp_fit_fun` and `curve_fit_wrapper` as static methods in `BaseAnalysis`. When other experiments (Ramsey, RB) are ready, we will decide if we want to apply further steps to prevent code duplication.